include/exadg/utilities/spatial_aware_function.h

```diff
@@ -0,0 +1,62 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2024 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef EXADG_AERO_UTILITIES_SPATIAL_AWARE_FUNCTION_H_
+#define EXADG_AERO_UTILITIES_SPATIAL_AWARE_FUNCTION_H_
+
+#include <deal.II/base/function.h>
+
+namespace ExaDG
+{
+namespace Utilities
+{
+/**
+ * This is a @c dealii::Function, that provides information if its result is constant over space at
+ * every time.
+ */
+template<int dim, typename Number = double>
+class SpatialAwareFunction : public dealii::Function<dim, Number>
+{
+public:
+  using time_type = typename dealii::Function<dim, Number>::time_type;
+
+  SpatialAwareFunction(const unsigned int n_components = 1, const time_type initial_time = 0.0)
+    : dealii::Function<dim, Number>(n_components, initial_time)
+  {
+  }
+
+  /**
+   * Varies in space, i.e. is not constant over space at every time.
+   */
+  virtual bool
```

Comment:
Please add the following documentation to this function:

"If this function evaluates to true, `dealii::Function::set_time()` in combination with `dealii::Function::value()` will be used to evaluate the function's value."

Reply:
[4776f56](https://github.com/exadg/exadg/pull/644/commits/4776f5622e28a7d092692b74ed7fe858ab539238)